docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,446 @@
+The backends for AMD GPUs must all be explicilty enabled.
+The tests are disabled, but the examples will still be built.
+
+
+Build oneMKL for the DFT domain only with support for x86 CPU, Intel GPU, AMD GPU and Nvidia GPU with
+testing enabled:
+
+.. code-block:: bash
+
+    cmake $ONEMKL_DIR \ 
+        -DCMAKE_CXX_COMPILER=icpx \
+        -DCMAKE_C_COMPILER=icx \ 
+        -DENABLE_ROCFFT_BACKEND=ON \
+        -DENABLE_CUFFT_BACKEND=ON \
+        -DTARGET_DOMAINS=dft \
+        -DBUILD_EXAMPLES=OFF
+
+Note that this is not a supported configuration, and requires Codeplay's oneAPI for 
+`AMD <https://developer.codeplay.com/products/oneapi/amd/home/>`_ and 
+`Nvidia <https://developer.codeplay.com/products/oneapi/nvidia/home/>`_ GPU plugins.
+Like the above example, the MKLCPU and MKLGPU backends are enabled by default, with
+backends for Nvidia GPU and AMD GPU explicitly enabled.
+``-DTARGET_DOMAINS=dft`` causes only DFT backends to be built. If this was not set,
+the backend libraries to enable the use of BLAS, LAPACK and RNG with MKLGPU and MKLCPU
+would also be enabled.
+The build of examples is disabled.
+Since functional testing was not disabled, tests would be built.
+
+.. _project_cleanup:
+
+Project Cleanup
+###############
+
+Most use-cases involve building the project without the need to cleanup the
+build directory. However, if you wish to cleanup the build directory, you can
+delete the ``build`` folder and create a new one. If you wish to cleanup the
+build files but retain the build configuration, following commands will help
+you do so.
+
+.. code-block:: sh
+
+   # If you use "GNU/Unix Makefiles" for building,
+   make clean
+
+   # If you use "Ninja" for building
+   ninja -t clean
+
+
+.. _build_for_windows_dpcpp:
+
+Building for Windows
+####################
+
+The Windows build is similar to the Linux build, albeit that 
+`fewer backends are supported <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#windows>`_.
+For example:
+
+.. code-block:: bash
+
+    # Inside <path to onemkl>
+    md build && cd build
+    cmake .. -G Ninja [-DCMAKE_CXX_COMPILER=<path_to_icx_compiler>\bin\icx]      # required only if icx is not found in environment variable PATH
+                      [-DCMAKE_C_COMPILER=<path_to_icx_compiler>\bin\icx]        # required only if icx is not found in environment variable PATH
+                      [-DMKL_ROOT=<mkl_install_prefix>]                          # required only if environment variable MKLROOT is not set
+                      [-DREF_BLAS_ROOT=<reference_blas_install_prefix>]          # required only for testing
+                      [-DREF_LAPACK_ROOT=<reference_lapack_install_prefix>]      # required only for testing
+    ninja
+    ctest
+    cmake --install . --prefix <path_to_install_dir>                             # required to have full package structure
+
+.. _build_common_problems_dpcpp:
+
+Common Problems
```

Comment:
"Common Problems" sounds like everyone will encounter them, maybe "Build FAQ"?

Reply:
Renamed in [7921f06](https://github.com/oneapi-src/oneMKL/pull/510/commits/7921f063967bde46bcefc0991231d6fe490ff2c0).